src/pudl/analysis/allocate_net_gen.py

```diff
@@ -21,46 +21,83 @@
 
-We allocate the net generation reported in the generation_fuel_eia923 table on the basis
-of plant, prime mover, and fuel type among the generators in each plant that have
-matching fuel types. Generation is allocated proportional to reported generation if it's
-available, and proportional to each generator's capacity if generation is not available.
-
-In more detail: within each year of data, we split the plants into three groups:
-
-* Plants where ALL generators report in the more granular generation_eia923
-  table.
-* Plants where NONE of the generators report in the generation_eia923 table.
-* Plants where only SOME of the generators report in the generation_eia923
-  table.
-
-In plant-years where ALL generators report more granular generation, the total net
-generation reported in the generation_fuel_eia923 table is allocated in proportion to
-the generation each generator reported in the generation_eia923 table. We do this
-instead of using net_generation_mwh from generation_eia923 because there are some small
-discrepancies between the total amounts of generation reported in these two tables.
-
-In plant-years where NONE of the generators report more granular generation, we create a
-generator record for each associated fuel type. Those records are merged with the
+There are six main stages of the allocation process in this module:
+
+* Inputs! Grabbing the input tables from :class:`pudl.output.pudl.PudlTabl`. Some
+  cleaning (like :func:`distribute_annually_reported_data_to_months_if_annual`) and
+  aggregation (like :func:`allocate_bf_data_to_gens`) happen in this stage.
+* Associating the data from the input tables with generator/prime mover/energy source
+  codes - using the primary keys in :py:const:`IDX_GENS_PM_ESC` (see
+  :func:`associate_generator_tables`).
+* Prepare for the allocation (see :func:`prep_alloction_fraction`) by adding boolean
+  flags for whether a particular :py:const:`IDX_GENS_PM_ESC` shows up in each of the
+  input data tables.
+* Allocation! Do the actual allocating. See :func:`allocate_net_gen_by_gen_esc` for net
+  generation allocation and :func:`allocate_fuel_by_gen_esc` for fuel allocation. More
+  details on the allocation process below.
+* Tests! The high-level test of whether the allocated data matches the original inputs
+  is :func:`test_the_original_gf_vs_the_allocated_by_gens_gf`. There are also other
+  warnings that are logged if assumptions aren't met via :func:`_test_frac`,
+  :func:`_test_gen_fuel_allocation` and :func:`_test_gen_pm_fuel_output`
+* Aggregate the main output with primary keys of :py:const:`IDX_GENS_PM_ESC` to the
+  generator-level with primary keys of :py:const:`IDX_GENS` via
+  :func:`aggregate_gen_fuel_by_generator`.
+
+MOAR High-level description about the allocaiton step:
+
+We allocate the net generation/fuel consumption reported in the
+generation_fuel_eia923/boiler_fuel_eia923 table on the basis of plant, prime mover,
```

Comment:
I've been trying to link any mention of specific tables to their data dictionary so folks can see what's in them if they want to, with e.g. `:ref:`generation_fuel_eia923`